test/Test/TreasuryDAO.hs

```diff
@@ -0,0 +1,186 @@
+      ]
+  ]
+
+validProposal :: (Monad m) => NettestImpl m -> m ()
+validProposal = uncapsNettest $ do
+  ((owner1, _), (owner2, _), dao, _) <-
+      originateBaseDaoWithBalance def config
+          (\owner1_ owner2_ ->
+              [ ((owner1_, DAO.unfrozenTokenId), 200)
+              , ((owner2_, DAO.unfrozenTokenId), 100)
+              ]
+          )
+
+  let proposal = tokenTransferProposalMetadata (toAddress dao) owner2 owner1 -- transfer token from owner2 -> owner1
+      params t = DAO.ProposeParams
+        { ppFrozenToken = t
+        , ppProposalMetadata = proposal
+        }
+      expectedToken = fromInteger $ toInteger $ length $ lPackValueRaw proposal
+
+  callFrom (AddressResolved owner1) dao (Call @"Propose") (params $ expectedToken - 1)
+    & expectCustomError_ #fAIL_PROPOSAL_CHECK
+
+  callFrom (AddressResolved owner1) dao (Call @"Propose") (params $ expectedToken + 1)
+    & expectCustomError_ #fAIL_PROPOSAL_CHECK
+
+  -- Expected token is 58 in this case
+  _ <- createSampleProposal expectedToken proposal owner1 dao
+
+  checkTokenBalance (DAO.frozenTokenId) dao owner1 117
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner1 83
+
+flushTokenTransfer :: (Monad m) => NettestImpl m -> m ()
+flushTokenTransfer = uncapsNettest $ do
+  ((owner1, _), (owner2, _), dao, admin) <-
+      originateBaseDaoWithBalance def config
+          (\owner1_ owner2_ ->
+              [ ((owner1_, DAO.unfrozenTokenId), 200)
+              , ((owner2_, DAO.unfrozenTokenId), 100)
+              ]
+          )
+
+  -- Set RegistryDAO as operator of the address that is mean to transfer the tokens
+  let opParams = FA2.OperatorParam
+        { opOwner = owner2
+        , opOperator = toAddress dao
+        , opTokenId = DAO.unfrozenTokenId
+        }
+  callFrom (AddressResolved owner2) dao (Call @"Update_operators") [FA2.AddOperator opParams]
+
+  let proposal = tokenTransferProposalMetadata (toAddress dao) owner2 owner1 -- transfer token from owner2 -> owner1
+      expectedToken = getTokensAmount proposal
+
+  key1 <- createSampleProposal expectedToken proposal owner1 dao
+
+  checkTokenBalance (DAO.frozenTokenId) dao owner1 117
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner1 83
+
+  let
+    upvote = DAO.NoPermit DAO.VoteParam
+        { vVoteType = True
+        , vVoteAmount = 1
+        , vProposalKey = key1
+        }
+
+  callFrom (AddressResolved owner2) dao (Call @"Vote") [upvote]
+  advanceTime (sec 20)
+  callFrom (AddressResolved admin) dao (Call @"Flush") Nothing
+
+  checkTokenBalance (DAO.frozenTokenId) dao owner1 0
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner1 210
+  checkTokenBalance (DAO.frozenTokenId) dao owner2 0
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner2 90
+
+flushXtzTransfer :: (Monad m) => NettestImpl m -> m ()
+flushXtzTransfer = uncapsNettest $ do
+  ((owner1, _), (owner2, _), dao, admin) <- originateBaseDaoWithConfig def config
+
+  sendXtz (toAddress dao)
```

Comment:
AFAIU we transfer to `%default` here. Is there a `%default` entrypoint?

Reply:
There is no default entrypoint. I'm waiting for this: `https://github.com/tqtezos/baseDAO/pull/88`

Should I try sending xtz to another entrypoint in the meantime ? (Have to remove this check: https://github.com/tqtezos/baseDAO/blob/master/src/Lorentz/Contracts/BaseDAO.hs#L103)

Reply:
No, I think it's ok to wait, I was just wondering how it works, but AFAIU it doesn't.

Reply:
I've now rebased on #88, the tests work fine now. 

2 things to note:

- I have to remove `ensureZeroTransfer` for now, since it affects the custom entrypoints as well which results in now allowing to transfer xtz.

- The current way to add custom entrypoints seem to need at least more than 2 entrypoints since it expects a sum-type.

So I have to declare it this way for now to make it work:
```
data TreasuryDaoExtraInterface
  = Default
  | None -- Not needed
```

Hope to get some advice on how to solve/improve this.

Reply:
> The current way to add custom entrypoints seem to need at least more than 2 entrypoints since it expects a sum-type.

Another option is using [this magic type wrapper](https://gitlab.com/morley-framework/morley/blob/martoon%2F%23513-custom-generics-ligo/code/lorentz/src/Lorentz/Entrypoints/Helpers.hs#L36) to make only one entrypoint work. 

Apparently its existence is pretty hard to detect, if you have ideas where it is worth mentioning it, please tell.